src/lib/operators/aggregate.cpp

```diff
@@ -440,6 +440,9 @@ void Aggregate::_aggregate() {
 
     const auto& hash_keys = keys_per_chunk[chunk_id];
 
+    // Sometimes, gcc is really bad at accessing loop conditions only once, so we cache that here.
+    const auto input_chunk_size = chunk_in->size();
```

Comment:
This reminds me of http://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop.
Just as a side note ...

Reply:
That's a good summary of the problem. We probably should include that in the seminar at some point